src/pages/ShopPage/components/HistoryContent/index.js

```diff
@@ -1,98 +1,140 @@
-            <HistoryParentCategoryList
-              parentsTitle="Products"
-              itemsList={products}
-              filtredCategories={useProductsCategories.data}
-            />
-            <Styles.EmptyDivToSpace />
-            <HistoryParentCategoryList
-              parentsTitle="Dishes"
-              itemsList={dishes}
-              filtredCategories={dishesCategories.data}
-            />
-            <Styles.EmptyDivToSpace />
-            <HistoryParentCategoryList
-              parentsTitle="SavedList"
-              itemsList={savedLists}
-              filtredCategories={useSavedListsCategories.data}
-            />
-          </Styles.Wrapper>
-        )
-      },
-    )
-
-    return CreateHistoryElement
+  const dispatch = useDispatch()
+  const [updateHistory, setUpdateHistory] = useState(false)
+
+  useEffect(() => {
+    dispatch(fetchGroceries())
+    dispatch(fetchProducts())
+    dispatch(fetchDishes())
+    dispatch(fetchSavedLists())
+  }, [dispatch])
+
+  useEffect(() => {
+    dispatch(fetchHistoryBasket())
+  }, [dispatch, updateHistory])
+
+  const groceriesCategoriesData = useSelector(selectGroceriesCategoriesData)
+  const groceriesStatus = useSelector(selectGroceriesStatus)
+
+  const productsCategories = useSelector(selectProductsCategoriesData)
+  const productsStatus = useSelector(selectProductsStatus)
+
+  const dishesCategories = useSelector(selectDishesCategoriesData)
+  const dishesStatus = useSelector(selectDishesStatus)
+
+  const savedListCategories = useSelector(selectSavedListCategoriesData)
+  const savedListStatus = useSelector(selectSavedListStatus)
+
+  const historyData = useSelector(selectHistoryData)
+  const historyStatus = useSelector(selectHistoryStatus)
+
+  const data = dataLoadingStatus([
+    groceriesStatus,
+    productsStatus,
+    dishesStatus,
+    savedListStatus,
+    historyStatus,
+  ])
+
+  if (data.isError) {
+    return 'Fetching data error...'
   }
+
+  if (data.isLoading) {
+    return 'Loading data...'
+  }
+
+  if (!data.isLoaded) {
+    return null
+  }
+
+  const showWhenEmpty = (
+    <Styles.Message>Your purchase history is empty!</Styles.Message>
+  )
+
+  if (historyData.length < 1) {
+    return showWhenEmpty
+  }
+
+  const CreateHistoryElement = historyData.map(
+    ({ groceries, products, dishes, savedLists, id, DateToShow }) => {
+      return (
+        <Styles.Wrapper key={id}>
+          <div>
+            <Styles.Date>{DateToShow}</Styles.Date>
+            <Styles.Button
+              onClick={() => {
+                dispatch(deleteHistoryBasket({ id: id }))
+                setUpdateHistory((old) => !old)
+              }}
+            >
+              Delete from history
+            </Styles.Button>
+          </div>
+
+          <HistoryParentCategoryList
+            parentsTitle="Groceries"
+            itemsList={groceries}
+            filtredCategories={groceriesCategoriesData}
+          />
+          <Styles.EmptyDivToSpace />
```

Comment:
zrób marginbottom na liście zamiast takiego czegoś - nie powinno się robić komponentów, jeżeli nie zawierają żadnych danych

Reply:
zrobiłem tak na początku ale problem tkwi w tym że chciałem użyć w aplikacji scrollintoView i gdy zaznaczałem do jakiego elementu to zawsze początek zasłania menu bo ma ono position absolute a w metodzie scrollintoView nie umiałem znaleźć w dokumentacji ze na przykład chcesz wyświetlić górną krawędź ekranu np 30px poniżej

Reply:
ok to ma sens :D

Reply:
Done.